mm2src/coins/lightning/ln_rpc.rs

```diff
@@ -1,61 +1,70 @@
-use bitcoin_hashes::Hash;
+#[cfg(not(target_arch = "wasm32"))] use bitcoin_hashes::Hash;
 use common::executor::spawn;
-use common::{block_on, log};
+use common::log;
 #[cfg(not(target_arch = "wasm32"))] use derive_more::Display;
 use futures::compat::Future01CompatExt;
-use keys::hash::H256;
-use lightning::chain::{chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator},
+#[cfg(not(target_arch = "wasm32"))] use keys::hash::H256;
+use lightning::chain::chaininterface::BroadcasterInterface;
+#[cfg(not(target_arch = "wasm32"))]
+use lightning::chain::{chaininterface::{ConfirmationTarget, FeeEstimator},
                        Filter, WatchedOutput};
+#[cfg(not(target_arch = "wasm32"))]
 use rpc::v1::types::H256 as H256Json;
-use std::convert::TryFrom;
+#[cfg(not(target_arch = "wasm32"))] use std::cmp;
+#[cfg(not(target_arch = "wasm32"))] use std::convert::TryFrom;
 
-impl FeeEstimator for UtxoStandardCoin {
+#[cfg(not(target_arch = "wasm32"))]
+impl FeeEstimator for PlatformFields {
     // Gets estimated satoshis of fee required per 1000 Weight-Units.
     fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
-        let conf = &self.as_ref().conf;
-        // TODO: Maybe default_fee and confirmation targets can be set in coin configs or lightning configs (would require to move lightning config to coin config) instead
+        let platform_coin = &self.platform_coin;
+
         let default_fee = match confirmation_target {
-            // fetch background feerate
-            ConfirmationTarget::Background => 253,
-            // fetch normal feerate (~6 blocks)
-            ConfirmationTarget::Normal => 2000,
-            // fetch high priority feerate
-            ConfirmationTarget::HighPriority => 5000,
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.default_feerate,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.default_feerate,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.default_feerate,
         } * 4;
 
+        let conf = &platform_coin.as_ref().conf;
         let n_blocks = match confirmation_target {
-            // fetch background feerate
-            ConfirmationTarget::Background => 12,
-            // fetch normal feerate (~6 blocks)
-            ConfirmationTarget::Normal => 6,
-            // fetch high priority feerate
-            ConfirmationTarget::HighPriority => 1,
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.n_blocks,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.n_blocks,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.n_blocks,
         };
-        let fee_per_kb = block_on(
-            self.as_ref()
+        let fee_per_kb = tokio::task::block_in_place(move || {
```

Comment:
It seems like a good choice to use `block_in_place`! I hope it will work as expected.

Reply:
I already tested this and it works great :)
Before using it, `block_on` was causing some issues so I tried using `.wait()` like it's used here https://github.com/KomodoPlatform/atomicDEX-API/blob/5506cd7aa37b233bd1dc3b02196bc8531e64731c/mm2src/coins/eth.rs#L2739 and in other places (maybe we should fix those too), and because of blocking a remote node was not able to open a channel with our node as `get_est_sat_per_1000_weight` would block our node responding to pings so the connection to our node would fall. Also, there were other issues in receiving payments and closing channels before using `block_in_place`. Now everything works more smoothly.